### PR TITLE
Enable disabling of field validation for CRD schemas

### PR DESCRIPTION
**Pull request by ilackarms**
this PR enables the ability to disable validation of specific fields in crd schemas.

this is done via a two step process:

1. user specifies the fields in their proto files for which proto validation should be skipped:
    ```proto
    import "encoding/protobuf/cue/cue.proto";

    message RecursiveType {
      // validation will be skipped for these fields (unknown fields will be allowed)
      RecursiveType recursive_field = 1 [(cue.opt).disable_openapi_validation = true];
      repeated RecursiveType repeated_recursive_field = 2 [(cue.opt).disable_openapi_validation = true];
    }
    ```

2. user provides a top-level Go mapping which lists the recursive fields of each type to the cue library's`openapi.Generator` parameters:

    ```go
    generator := &openapi.Generator{
    		// k8s structural schemas do not allow $refs, i.e. all references must be expanded
    		ExpandReferences:   true,
               // ensures the given fields are marked as unstructured
    		UnstructuredFields: map[string][][]string{
               // maps the **go package** of the proto types to the **full json path** of the recursive fields within the CRD
    		"github.com/solo-io/skv2/codegen/test/api/things.test.io/v1": {
    			{"PaintSpec", "recursiveType", "recursiveField"},
    			{"PaintSpec", "recursiveType", "repeatedRecursiveField"},
    		},
    	},
    }
    ``` 

it should be possible for consuming code to generate the `UnstructuredFields` map based on parsing of the proto options; currently this is not implemented due to additional complexity and it is out of scope for Cue itself (more likely to be implemented in SKv2)

### CODE REVIEW

**Comment by ilackarms** [comment]
the output yaml in the crd's openapi schema will look like this:

```yaml
            recursiveType:
              description: OpenAPI gen test for recursive fields
              properties:
                recursiveField:
                  type: object
                  x-kubernetes-preserve-unknown-fields: true
                repeatedRecursiveField:
                  items:
                    type: object
                    x-kubernetes-preserve-unknown-fields: true
                  type: array
```

File: encoding/openapi/crd.go [hunk 1]
```diff
@@ -63,6 +63,34 @@ func (b *builder) coreSchemaWithName(name string) *ast.StructLit {
 	return s
 }
 
+func (b *builder) isUnstructured() bool {
+	path := b.ctx.path
+	if len(path) == 0 {
+		return false
+	}
+	if path[len(path)-1] == "*" {
```
Comment (harveyxia) [inline encoding/openapi/crd.go]: why are we stripping trailing asterisks?

File: encoding/protobuf/cue/cue.proto [hunk 2]
```diff
@@ -4,11 +4,15 @@ package cue;
...
 import "google/protobuf/descriptor.proto";
 
-option go_package = "cuelang.org/cueproto";
+option go_package = "cuelang.org/go/encoding/protobuf/cue";
 option java_package = "org.cuelang.cueproto";
 
 message FieldOptions {
     bool required = 1;
+
+    // Enable this option to treat this field use an unstructured object in the OpenAPI schema for this field.
```
Comment (harveyxia) [inline encoding/protobuf/cue/cue.proto]:
```suggestion
    // Enable this option to treat this field as an unstructured object in the OpenAPI schema.
```

File: encoding/openapi/crd.go (hunk 1 above)
Comment (ilackarms) [reply encoding/openapi/crd.go]: this is for the case when the field is repeated. i'll add a comment

**Review by harveyxia — APPROVED** [review]

**Pull request merged**